pyopencl/array.py

```diff
@@ -705,6 +736,63 @@ def __init__(
                          "than expected, potentially leading to crashes.",
                          InconsistentOpenCLQueueWarning, stacklevel=2)
 
+    # {{{ Pickling
+
+    def __getstate__(self):
```

Comment:
I think it'd be useful if  this worked for subclasses (liked `TaggedCLArray`), too. Should be tested, too.